Solutions/Marain.UserNotifications.Abstractions/Marain/NotificationTemplates/ITenantedNotificationTemplateStoreFactory.cs

```diff
@@ -0,0 +1,22 @@
+﻿// <copyright file="ITenantedNotificationTemplateStoreFactory.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.NotificationTemplates
+{
+    using System.Threading.Tasks;
+    using Corvus.Tenancy;
+
+    /// <summary>
+    /// Interface for a factory that can create a tenant-specific <see cref="INotificationTemplateStore"/>.
+    /// </summary>
+    public interface ITenantedNotificationTemplateStoreFactory
+    {
+        /// <summary>
+        /// Retrieves a <see cref="INotificationTemplateStore"/> for the specified <see cref="ITenant"/>.
+        /// </summary>
+        /// <param name="tenant">The tenant to retrieve the store for.</param>
+        /// <returns>The template store for the tenant.</returns>
+        Task<INotificationTemplateStore> GetTemplateStoreForTenantAsync(ITenant tenant);
```

Comment:
This pattern was copied from the existing code base from class `TenantedAzureTableUserNotificationStoreFactory` which creates a tenanted azure table. This also does not account for returning null either. 

So in our case, if the template store for that tenant is not found, the underlying code will create one and return the `NotificationTemplateStore`.

Reply:
We've created this issue - https://github.com/corvus-dotnet/Corvus.Tenancy/issues/267 - because we've realised that the concern raised doesn't just apply to this PR, it applies everywhere we've used this pattern. So we'll resolve this conversation and deal with the other issue separately.